[마켓불리 김원희] Modify: 장바구니에 데코레이터 적용 및 사용자 선택 DB에 저장하도록 수정

supersaloon opened the pull request:
>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
- 장바구니에 인증 데코레이터 적용하여 사용자 인증 하도록 수정
- 장바구니에서 사용자가 선택한 상품 정보를 DB에 저장하도록 수정
		 
## 수정 사항들 자세한 내용
- 장바구니에 인증 데코레이터 적용하여 사용자 인증 하도록 수정
  - 기존 데코레이터 구현 전에 사용자를 임의로 지정하였으나, 인증데코레이터로 사용자 인증 받도록 수정 
- 장바구니에서 사용자가 선택한 상품 정보를 DB에 저장하도록 수정
  - 장바구니에서 사용자가 선택 / 선택 해제하면 Cart 테이블에 is_selected로 저장하도록 구현


## 기타 질문 및 특이 사항


## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

Commit:
Modify: 장바구니 상품 선택 기능 구현
- 장바구니 상품을 선택하면 Cart 테이블의 is_selected 에 그 상태를 저장

Commit:
Modify: 장바구니에 데코레이터 적용하여 사용자 인증 받도록 구현

jamessoun93 commented on order/views.py:
```diff
@@ -8,15 +8,16 @@
         try:
             data = json.loads(request.body)
-            user_id = "1"
+            user_id = str(request.user.id)
```
문자열로 형변환해주실 필요 없으시고, `user_id` 라는 변수에 굳이 담아주실 필요없이 필요한 부분에서 바로 `request.user.id` 로 활용하시면 될 것 같네요!

supersaloon replied:
사용자 id를 request.user.id로 바로 꺼내어 사용하도록 수정하였습니다.

더불어, str 형 변환을 한 이유는  
`new_order.order_number = user_id + "-" + datetime.now().strftime('%Y-%m-%d-%H-%M-%S-%f')` 이부분에서 int인 user_id 에 대해 string 의 + 연산을 시도하면 에러가 발생하여서 였습니다. 

이부분 코드를 `new_order.order_number = f"{request.user.id}-{datetime.now().strftime('%Y-%m-%d-%H-%M-%S-%f')}"` 로 수정하여 형변환 없이 string 만들수 있도록 하였습니다.

jamessoun93 replied:
아 주문번호 생성로직 부분을 캐치를 못했군요 제가! 확인했습니다~ 👍

Commit:
Modify: PR #13 에 대한 리뷰 반영하여 수정
- `user_id`를 변수에 닫지 않고 바로 `request`에서 꺼내어 사용하도록 수정
- typo 수정(`if not flag` -> `if flag`)

jamessoun93 commented:
LGTM
